js/viz/core/helpers.js

```diff
@@ -61,14 +61,53 @@ function addChange(settings) {
     buildTotalChanges(proto);
 }
 
+function empty() { }
+
+function createChainExecutor() {
+    var chain = [];
+
+    executeChain.add = function(item) { chain.push(item); };
+    return executeChain;
+
+    function executeChain() {
+        var i,
+            ii = chain.length;
+        for(i = 0; i < ii; ++i) {
+            chain[i].apply(this, arguments);
+        }
+    }
+}
+
+function expand(target, name, expander) {
+    var current = target[name];
+    if(current.add) {
+        current.add(expander);
+    } else if(current === empty) {
```

Comment:
I don't like that we should reference and use this particular `empty` function to implement expandable members. Can we re-work it?